CONTRIBUTING.md

```diff
@@ -1,35 +1,199 @@
+
+Any issue opened without effort to provide the required details for developers, contributors or anybody else to reproduce the problem will be closed as invalid.
+For example:
+-   Crash reports with just a stack trace;
+-   Speculated performance issues that do not come with actual profiling data + analysis supporting the claim;
+
+## Before submitting a bug report
+
+-   **Do some basic troubleshooting**:
+    -   restart qBittorrent;
+    -   restart your PC;
+    -   update your OS (e.g. Windows updates);
+    -   fully reinstall qBittorrent;
+    -   etc...
+-   Make sure the problem is not caused by anti-virus or other program messing with your files;
+-   Check if you can reproduce the problem in the latest version of qBittorrent;
+-   **Check forum and wiki**. You might be able to find the cause of the problem and fix things yourself;
+-   **Check if the issue exists already in the issue tracker**.
+    -   If it does and the issue is still open, add a comment to the existing issue instead of opening a new one;
+    -   If you find a Closed issue that seems like it is the same thing that you're experiencing, open a new issue and include a link to the original issue in the body of your new one.
+-   If the issue is with the search functionality:
+    -   make sure you have `python` installed correctly (remember the search functionality requires a working python installation);
+    -   make sure it is in fact a problem with the search functionality itself, and not a problem with the plugins. If something does not work properly with the search functionality, the first step is to rule out search plugin-related issues.
+        -   For search plugin issues, report on the respective search plugin support page, or at qbittorrent/search-plugins.
+
+## Steps to ensure a good bug report
+
+**Follow these guidelines** in order to provide as much useful information as possible right away.
+Otherwise, we've noticed that a lot of your time (and the developers') gets thrown away on exchanging back and forth to get this information.
+
+
+*   Use a **clear and descriptive title** for the issue to identify the problem.
+
+*   Post only **one specific issue per submission**.
+
+*   **Fill out the issue template properly**.
+
+-   **Make sure you are using qBittorrent on a supported platform**. Do not submit issues which can only be reproduced on beta/unsupported releases of supported operating systems (e.g. Windows 10 Insider, Ubuntu 12.04 LTS, etc).
+These are unstable/unsupported platforms, and in all likelihood, whatever the issue is, it is not related to qBittorrent.
+
+*   **Specify the OS you're using, its version and architecture**;
+    *   Examples: Windows 8.1 32-bit, Linux Mint 17.1 64-bit, Windows 10 Fall creators Update 64-bit, etc.
+
+
+*   **Report only if you can reproduce the issue in an official stable release, or in a beta release of qBittorrent** (beta testing is encouraged :smile:).
```

Comment:
Disagree. The reports about bugs in upstream are also important.

Reply:
Ok, how about:
> **Report only if you can reproduce the issue with an official stable release, a beta release, or with the most recent upstream changes (in this last case specify the specific commit you are on).** Beta testing is encouraged :smile:.